include/envoy/http/header_map.h

```diff
@@ -125,16 +124,23 @@ class HeaderString {
   void append(const char* data, uint32_t size);
 
   /**
-   * @return the modifiable backing buffer (either inline or heap allocated).
+   * Transforms the inlined vector data using the given UnaryOperation (conforms
+   * to std::transform).
+   * @param unary_op the operations to be performed on each of the elements.
    */
-  char* buffer() { return buffer_.dynamic_; }
+  template <typename UnaryOperation> void inlineTransform(UnaryOperation&& unary_op) {
```

Comment:
I would suggest adding this method in to the fuzz test to replicate the header manipulation in the codec. Perhaps here you can add an inlineTransform:
https://github.com/envoyproxy/envoy/blob/caf3896a7b4b4acdb1519ef02ec27d36e01f0f43/test/common/http/header_map_impl_fuzz_test.cc#L123

Reply:
Added, as is being done here:
https://github.com/envoyproxy/envoy/blob/f7de5ebdb2459250f4fde233bc0c73d5c5336967/source/common/http/http1/codec_impl.cc#L393-L396